### PR TITLE
Add DiscussionEvent webhook (google#2221)

### LINKED ISSUE #2221: Support Discussions webhook event
GitHub Discussions was out of beta in August 2021 (https://github.blog/2021-08-17-github-discussions-out-of-beta/).

Webhook event document is here: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion

### PR DESCRIPTION
**Pull request by maniSbindra**
This commit has changes to enable support to receive Discussions Webhook Events, to fix #2221.
Changes include:
- Addition of event type for this type of event (DiscussionEvent)
- Changes for identification and mapping of DiscussionEvent
- Addition / Modification of relevant tests
- Addition of files created by go generate

### CODE REVIEW

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:209]:
```suggestion
	Number             *int              `json:"number,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @maniSbindra !
Just a few minor tweaks, please.

**Comment by maniSbindra** [comment]
> Thank you, @maniSbindra ! Just a few minor tweaks, please.

Thanks @gmlewis , I have incorporated the changes, request you to please review.

File: github/messages_test.go
Comment (gmlewis) [inline github/messages_test.go:300]: Please move this new block a few lines down, after `DeploymentStatusEvent` to keep them sorted.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @maniSbindra !
Just one minor tweak, please, then we should be ready for a second LGTM before merging.

Note that any other contributor to this repo is welcome to provide the second approval and LGTM.  Thank you!

File: github/messages_test.go
Comment (maniSbindra) [reply github/messages_test.go:300]: Done.

**Review by gmlewis — APPROVED** [review]
Thank you, @maniSbindra !
LGTM.

Awaiting second LGTM from any other contributor to this repo.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**
